### ISSUE TITLE
Create SimpleRegressor class

### ISSUE THREAD

**Issue by skekre98** [skekre98/simple_learn#4]
**Description of feature/enhancement**
This class should handle generating the most accurate model for for any given regression problem.

**Description of implementation**
The class should have a function called `fit` which creates the most accurate regression model for any given data set.

**Why this is a good feature/enhancement**
This will allow users to simply pass in data and be returned the most accurate model without having to worry about fine tuning the hyper-parameters themselves.

**Additional context**
n/a

**Comment by eranns** [comment]
Hey @skekre98 I would like to help with this issue and writing tests for it while i'm writing the class ( #11 ).
I thought about adding the following regressors to `param_grid` (in another dictionary) for a start : 

"SGDRegressor"
"KNeighborsRegressor"
"DecisionTreeRegressor"
"RandomForestRegressor"
"GradientBoostingRegressor"
"AdaBoostRegressor"

They seem to cover easier to harder cases of data. Then We can see how they work out and improve from there.

Let me know what you think.

**Comment by skekre98** [comment]
Hi @eranns, sorry about my absence. I think it would be a great idea to start with these regressors. Instead of adding the regressor `param_map` to the current file, let's go ahead and create a separate directory for regressors and add another file called `param_grid.py` there. 

I'll go ahead and assign this and #11 over to you 😄

**Comment by eranns** [comment]
> Hi @eranns, sorry about my absence. I think it would be a great idea to start with these regressors. Instead of adding the regressor `param_map` to the current file, let's go ahead and create a separate directory for regressors and add another file called `param_grid.py` there.
> 
> I'll go ahead and assign this and #11 over to you 😄

alright I created separate directory for regressors.

I wrote the class and the tests and made it similar to the classifiers code.  
There are few things I changed  : 
* metrics and that does not apply to regression but to classifications 
* added raise exception in gridsearch once model failed and used try/except
* gridsearch scoring is rmse (can choose another) but gridsearch try to maximize this value so instead it returns it as negative so I handled it

I will make a pull request so we can discuss on the code in a better way.
